src/App.tsx

```diff
@@ -1,26 +1,259 @@
-/* eslint-disable max-len */
+/* eslint-disable no-console */
+/* eslint-disable jsx-a11y/label-has-associated-control */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
-import { UserWarning } from './UserWarning';
+import React, { useEffect, useRef, useState } from 'react';
+import { deleteTodo, getTodos, addTodo, USER_ID } from './api/todos';
+import { Todo } from './types/Todo';
+import { Header } from './components/Header';
+import { TodoList } from './components/TodoList';
+import { Footer } from './components/Footer';
+import { Error } from './components/Errors';
 
-const USER_ID = 0;
+export enum Filter {
+  All = 'all',
+  Active = 'active',
+  Completed = 'completed',
+}
 
 export const App: React.FC = () => {
-  if (!USER_ID) {
-    return <UserWarning />;
+  const [query, setQuery] = useState<string>('');
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [itemsLeft, setItemsLeft] = useState<number>(0);
+  const [sortBy, setSortBy] = useState('all');
+  const [errorMessage, setErrorMessage] = useState<string | null | boolean>(
+    null,
+  );
+  const [deleteTodoId, setDeleteTodoId] = useState<number | null>(null);
+  const [isLoadingChange, setIsLoadingChange] = useState(false);
+  const [cleanCompleted, setCleanCompleted] = useState<boolean>(false);
+  const [newTask, setNewTask] = useState<string>('');
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [isSubmiting, setIsSubmiting] = useState<boolean>(false);
```

Comment:
1) Create enum for all the error `FiltersType.Default = ''`
2) Create enum for statuses
3) if u set (string, number, logical value) as the initial value - u don't need a generic type for this
```suggestion
  const [query, setQuery] = useState('');
  const [todos, setTodos] = useState<Todo[]>([]);
  const [itemsLeft, setItemsLeft] = useState(0);
  const [sortBy, setSortBy] = useState<FiltersType>(FiltersType.All);
  const [errorMessage, setErrorMessage] = useState<ErrorType>(ErrorType.Default);
  const [deleteTodoId, setDeleteTodoId] = useState<number | null>(null);
  const [isLoadingChange, setIsLoadingChange] = useState(false);
  const [cleanCompleted, setCleanCompleted] = useState(false);
  const [newTask, setNewTask] = useState('');
  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
  const [isSubmiting, setIsSubmiting] = useState(false);
```